lib/phoenix_live_view/upload_writer.ex

```diff
@@ -0,0 +1,32 @@
+defmodule Phoenix.LiveView.UploadWriter do
+
+  @callback init(opts :: term) :: {:ok, state :: term} | {:error, term}
+  @callback meta(state :: term) :: map
+  @callback write_chunk(state :: term, data :: binary) :: {:ok, state :: term} | {:error, term}
```

Comment:
Two questions, both pedantic:

1. Should the order of the arguments be flipped? i.e. `(data :: binary, state :: term)`?
2. Should we rename to `handle_chunk`?

```suggestion
  @callback handle_chunk(data :: binary, state :: term) :: {:ok, state :: term} | {:error, term}
```

Reply:
definitely flip the arguments. Naming not sure. It is called a "writer", so I am not sure about handle_chunk, but now I am having an existential crisis about it :D

Reply:
`write_chunk` is fine, no existential crisis necessary! 🤣 i'll update the suggestion